src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java

```diff
@@ -80,11 +80,12 @@ private SearchResultViewColumn createSearchResultViewColumn(final SearchResultFi
     }
 
     private boolean filterDistinctFieldsByRole(final HashSet<String> addedFields, final SearchResultField resultField) {
-        if (addedFields.contains(resultField.getCaseFieldId())) {
+        String id = buildCaseFieldId(resultField);
+        if (addedFields.contains(id)) {
             return false;
         } else {
             if (StringUtils.isEmpty(resultField.getRole()) || userRepository.getUserRoles().contains(resultField.getRole())) {
```

Comment:
just realised that this is not gonna work for case roles. Nothing to do with this PR but raised the following ticket to keep track of this: https://tools.hmcts.net/jira/browse/RDM-7942